spigot-impl/src/main/java/me/aurium/beetle/spigot/SpigotOptions.java

```diff
@@ -0,0 +1,104 @@
+package me.aurium.beetle.spigot;
+
+import me.aurium.beetle.api.BeetleOptions;
+import me.aurium.beetle.api.task.sync.BlockingBehavior;
+import me.aurium.beetle.defaults.task.sync.TimerSleepBehavior;
+
+import java.util.Objects;
+import java.util.concurrent.Executor;
+
+public class SpigotOptions implements BeetleOptions {
+
+    private final boolean isDebug;
+    private final boolean usePlatformExecutor;
+    private final boolean useThisThread;
+
+    private final Executor platformExecutor;
+    private final Thread mainThread;
+    private final BlockingBehavior blockingBehavior;
+
+    public SpigotOptions(boolean isDebug, boolean usePlatformExecutor, boolean useThisThread, Executor platformExecutor, Thread mainThread, BlockingBehavior blockingBehavior) {
+        this.isDebug = isDebug;
+        this.usePlatformExecutor = usePlatformExecutor;
+        this.useThisThread = useThisThread;
+        this.platformExecutor = platformExecutor;
+        this.mainThread = mainThread;
+        this.blockingBehavior = blockingBehavior;
+    }
+
+    public boolean isDebug() {
+        return isDebug;
+    }
+    public Executor getPlatformExecutor() {
+        return platformExecutor;
+    }
+    public boolean isUsePlatformExecutor() {
+        return usePlatformExecutor;
+    }
+    public boolean isUseThisThread() {
+        return useThisThread;
+    }
+    public Thread getMainThread() {
+        return mainThread;
+    }
+
+    public BlockingBehavior getBlockingBehavior() {
+        return blockingBehavior;
+    }
+
+    public static class Builder {
+        private boolean isDebug;
+        private boolean usePlatformExecutor;
+        private boolean useThisThread;
+
+        private Executor platformExecutor;
+        private Thread mainThread;
+        private BlockingBehavior blockingBehavior;
+
+        public Builder() {
+            isDebug = false;
+            usePlatformExecutor = false;
```

Comment:
You don't need to explicitly initialize variables to their default values if those defaults are Java's defaults (false, 0, null)

Reply:
I know, theyre there to remind me of what the values are.